### PR TITLE
support hyphens / emdash for provision ref ranges

### PR DESCRIPTION
**Pull request by longhotsummer**

### CODE REVIEW

File: indigo/analysis/refs/provision_refs.peg [hunk 1]
```diff
@@ -104,7 +104,8 @@ grammar ProvisionRefs
   or_af       <- or_en # `of`
 
                  # note that "tot" must come before "to" because the latter is a substring of the former
-  to          <- to_af / to_en
+  to          <- dash / to_af / to_en
+  dash        <- `-` / `—`
```
Comment (longhotsummer) [inline indigo/analysis/refs/provision_refs.peg:108]: @goose-life pls add your favourite dash / hyphen / minus variants

Comment (goose-life) [inline indigo/analysis/refs/provision_refs.peg:108]:
```suggestion
  dash        <- `-` / `–` / `—`
```

File: indigo/tests/test_provision_refs.py [hunk 2]
```diff
@@ -1442,6 +1442,24 @@ def test_mixed(self):
         ], result.references)
         self.assertIsNone(result.target)
 
+    def test_range(self):
+        result = parse_provision_refs("Section 1.2(1) to (3), (ii) — (iii), (g) - (j)")
```
Comment (goose-life) [inline indigo/tests/test_provision_refs.py:1446]:
```suggestion
        result = parse_provision_refs("Section 1.2(1) to (3), (ii) — (iii), (g)- (j)")
```

**Comment by longhotsummer** [comment]
It looks like we're not robust enough to support `(g)- (j)` rather than just `(g) - (j)` so I'm going to back out that change and open a separate issue.

**Comment by longhotsummer** [comment]
Created https://github.com/laws-africa/indigo/issues/1884 to track the range issue

**Pull request merged**